js/index.js

```diff
@@ -0,0 +1,235 @@
+  }
+  if (/(active)/.exec(window.location.href)) viewTodo();
+  else if (/(completed)/.exec(window.location.href)) viewDone();
+}
+
+function workDelete(e) {  // 등록된 항목들을 제거하는 기능
+  if(confirm("정말 삭제하시겠습니까?")){
+    let li = e.target.parentNode.parentNode;
+  console.log(li);
+  console.log(li.parentNode);
+  li.parentNode.removeChild(li);
+  renewStrong();
+  }
+}
+
+function workContentCopy(e) {   // 등록된 항목의 수정을 위해 내용을 입력칸에 복사하는 기능
+  let li = e.target.parentNode.parentNode;
+  li.classList.add("editing");
+  let chginput = e.target.parentNode.nextSibling;
+  chginput.value = e.target.innerText;
+}
+
+function workUpdate(e) {      // 등록된 항목을 수정하는 기능
+  let li = e.target.parentNode;
+  if (e.keyCode == 27) {
+    li.classList.remove("editing");
+  }
+  if (e.keyCode == 13) {
+    if (e.target.value !== "" && !/^\s+|\s+$/g.exec(e.target.value)) {
+      let label = e.target.previousSibling.childNodes[1];
+      label.innerText = e.target.value;
+      e.target.value = "";
+      li.classList.remove("editing");
+    } else {
+      alert("불필요한 공백을 제거해주세요!");
+    }
+  }
+}
+
+function renewStrong() {      // 리스트 하단의 총 목록 갯수를 갱신하는 기능
+  let list = document.querySelectorAll("#todo-list>li.selected");
+  let items = document.querySelector("strong");
+  items.innerText = list.length;
+}
+
+function viewAll() {    // "전체보기" 버튼 클릭 시의 기능
+  let list = document.querySelectorAll("#todo-list>li");
+  for (let i = 0; i < list.length; i++) {
+    list[i].classList.add("selected");
+  }
+  changeBox(viewAllList);
+  reflectView();
+}
+
+function viewTodo() {     // "해야할 일" 버튼 클릭 시의 기능
+  let list = document.querySelectorAll("#todo-list>li");
+
+  for (let i = 0; i < list.length; i++) {
+    if (list[i].firstChild.firstChild.hasAttribute("checked")) {
+      list[i].classList.remove("selected");
+    } else {
+      list[i].classList.add("selected");
+    }
+  }
+  changeBox(viewTodoList);
+  reflectView();
+}
+
+function viewDone() {       // "완료한 일" 버튼 클릭 시의 기능
+  let list = document.querySelectorAll("#todo-list>li");
+  for (let i = 0; i < list.length; i++) {
+    if (!list[i].firstChild.firstChild.hasAttribute("checked")) {
+      list[i].classList.remove("selected");
+    } else {
+      list[i].classList.add("selected");
+    }
+  }
+  changeBox(viewCompleteList);
+  reflectView();
+}
+
+function changeBox(box) {   // 선택한 버튼을 표시하는 기능
+  console.log(box);
+  if (box.classList.contains("all")) {
+    viewAllList.classList.add("selected");
+    viewTodoList.classList.remove("selected");
+    viewCompleteList.classList.remove("selected");
+  } else if (box.classList.contains("active")) {
+    viewAllList.classList.remove("selected");
+    viewTodoList.classList.add("selected");
+    viewCompleteList.classList.remove("selected");
+  } else if (box.classList.contains("completed")) {
+    viewAllList.classList.remove("selected");
+    viewTodoList.classList.remove("selected");
+    viewCompleteList.classList.add("selected");
+  }
```

Comment:
```suggestion
function changeBox(box) {
  viewAllList.classList.remove("selected");
  viewTodoList.classList.remove("selected");
  viewCompleteList.classList.remove("selected");
  
  if (box.classList.contains("all")) {
    viewAllList.classList.add("selected");
  } else if (box.classList.contains("active")) {
    viewTodoList.classList.add("selected");
  } else if (box.classList.contains("completed")) {
    viewCompleteList.classList.add("selected");
  }
}
```
이 부분은 비슷한 코드가 여러번 들어가서 가독성이 좋지 않았네요. 이런식으로 전체 List들의 ```selected``` 속성을 먼저 제거하고 box의 클래스를 판별한 다음 ```selected``` 속성을 추가하는게 더 좋을거 같습니다! 

```className```으로 판별하지 않고 ```classList.contains()``` 메서드를 사용한 부분은 생각하지 못 했던 부분인데 배워갑니다ㅎㅎ👍👍👍

Reply:
중복 제거의 중요성을 알면서도 이런 단순한 부분에서 깊은 생각을 하지 못하고 지나갔었다는게 충격적었습니다.  반복문이나 조건문에서 이런 부분에 대해 더 깊이있게 고려해야겠다고 느낄 수 있었던 좋은 지적 감사합니다.    
review#1 에 반영하여 수정하였습니다.